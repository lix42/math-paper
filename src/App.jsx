// @flow
import * as React from "react";
import { BrowserRouter, Link } from "react-router-dom";
import Routes from "./Routes";
import classes from "./styles";
import reactLogo from "./assets/React-icon.png";

const App = (): React.Node => (
  <BrowserRouter>
    <main className={classes.container}>
      <div>
        <h1>hello world!</h1>
        <img className={classes.image} alt="react logo" src={reactLogo} />
        <p>If you see this everything is working!</p>
      </div>
      <ul className={classes.left}>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/about">About</Link>
        </li>
      </ul>
      <Routes />
    </main>
  </BrowserRouter>
);

export default App;
