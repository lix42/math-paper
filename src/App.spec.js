import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import App from "./App";

describe("App", () => {
  test("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  test("match snapshot", () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
