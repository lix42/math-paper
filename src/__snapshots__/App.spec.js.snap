// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App match snapshot 1`] = `
<BrowserRouter>
  <main
    className="container-0-0-1"
  >
    <div>
      <h1>
        hello world!
      </h1>
      <img
        alt="react logo"
        src="test-file-stub"
      />
      <p>
        If you see this everything is working!
      </p>
    </div>
    <ul
      className="left-0-0-2"
    >
      <li>
        <Link
          replace={false}
          to="/"
        >
          Home
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/about"
        >
          About
        </Link>
      </li>
    </ul>
    <Routes />
  </main>
</BrowserRouter>
`;
